src/server/terrain/structure_building_blocks.zig

```diff
@@ -0,0 +1,290 @@
+	}
+	if(zon.array.items.len == 0) {
+		std.log.warn("['{s}'->'{s}'] Empty children list.", .{parentId, colorName});
+		return .init(arenaAllocator, &[0]Child{});
+	}
+	var list = arenaAllocator.alloc(Child, zon.array.items.len);
+	for(zon.array.items, 0..) |entry, childIndex| {
+		list[childIndex] = Child.initFromZon(parentId, colorName, colorIndex, childIndex, entry);
+	}
+	return .init(arenaAllocator, list);
+}
+
+const Child = struct {
+	structure: *StructureBuildingBlock,
+	chance: f32,
+
+	fn initFromZon(parentId: []const u8, colorName: []const u8, colorIndex: usize, childIndex: usize, zon: ZonElement) Child {
+		const structureId = zon.get([]const u8, "structure", "");
+		if(structureId.len == 0) {
+			std.log.warn("['{s}'->'{s}'->'{d}'] Child node has empty structure field.", .{parentId, colorName, childIndex});
+		}
+		childrenToResolve.append(.{.parentId = parentId, .colorName = colorName, .colorIndex = colorIndex, .childIndex = childIndex, .structureId = structureId});
+		return .{
+			.structure = undefined,
+			.chance = zon.get(f32, "chance", 1.0),
+		};
+	}
+};
+
+pub fn registerSBB(structures: *std.StringHashMap(ZonElement)) !void {
+	if(structureCache.capacity() != 0) {
+		std.log.err("Attempting to register new SBBs without resetting cache.", .{});
+		return error.AlreadyRegistered;
+	}
+	std.log.debug("Registering {} structure building blocks", .{structures.count()});
+	structureCache.ensureTotalCapacity(arenaAllocator.allocator, structures.count()) catch unreachable;
+	childrenToResolve = .init(main.stackAllocator);
+	{
+		var iterator = structures.iterator();
+		while(iterator.next()) |entry| {
+			const value = StructureBuildingBlock.initFromZon(entry.key_ptr.*, entry.value_ptr.*) catch continue;
+			const key = arenaAllocator.dupe(u8, entry.key_ptr.*);
+			const result = structureCache.getOrPut(arenaAllocator.allocator, key) catch unreachable;
+
+			if(result.found_existing) {
+				std.log.err("Ignoring duplicated structure building block: '{s}'", .{entry.key_ptr.*});
+				continue;
+			}
+			result.value_ptr.* = value;
+			std.log.debug("Registered structure building block: '{s}'", .{entry.key_ptr.*});
+		}
+	}
+	{
+		for(childrenToResolve.items) |entry| {
+			const parent = structureCache.getPtr(entry.parentId) orelse {
+				std.log.err("Could not find parent structure '{s}' for child resolution.", .{entry.structureId});
+				continue;
+			};
+			const child = structureCache.getPtr(entry.structureId) orelse {
+				std.log.err("Could not find child structure '{s}' for child resolution.", .{entry.structureId});
+				continue;
+			};
+
+			std.log.debug("Resolving child structure '{s}'->'{s}'->'{d}' to '{s}'", .{entry.parentId, entry.colorName, entry.childIndex, entry.structureId});
+			parent.children[entry.colorIndex].items[entry.childIndex].structure = child;
+			continue;
+		}
+		childrenToResolve.deinit();
+	}
+	std.log.debug("Registered {} structure building blocks", .{structureCache.count()});
```

Comment:
So it's printing the count 3 times? Twice here and once in assets.zig?
I think we should only print it once.